packages/wonder-blocks-data/components/data.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import * as React from "react";
+
+import {ResponseCache} from "../util/response-cache.js";
+import InternalData from "./internal-data.js";
+
+import InterceptContext from "./intercept-context.js";
+
+import type {
+    CacheEntry,
+    Interceptor,
+    Result,
+    IRequestHandler,
+} from "../util/types.js";
+
+type Props<
+    /**
+     * The type of options that the handler requires to define a request.
+     */
+    TOptions,
+    /**
+     * The type of data resolved by the handler's fulfillRequest method.
+     */
+    TData,
+> = {|
+    /**
+     * An `IRequestHandler` instance of the type this component will use to
+     * resolve its requests.
+     *
+     * The framework deduplicates handlers based on their `type` property.
+     * Handlers with the same `type` property are assumed to be the same.
+     */
+    handler: IRequestHandler<TOptions, TData>,
+
+    /**
+     * The handler-specific options that define what requestt is to be made.
+     *
+     * Changing these options will only cause the data to update if the key
+     * from `handler.getKey(options)` changes.
+     */
+    options: TOptions,
+
+    /**
+     * A function that will render the content of this component using the
+     * loading state and data or error that gets retrieved from cache or loaded
+     * via the request if no cached value is available.
+     */
+    children: (result: Result<TData>) => React.Node,
+|};
+
+/**
+ * We have the ability to override handlers and the cache for testing and such
+ * so we export this component, and it then wraps the real data component.
+ */
+export default class Data<TOptions, TData> extends React.Component<
+    Props<TOptions, TData>,
+> {
+    _getHandlerFromInterceptor(
+        interceptor: ?Interceptor,
+    ): IRequestHandler<TOptions, TData> {
+        const {handler} = this.props;
+        if (!interceptor) {
+            return handler;
+        }
+
+        const {fulfillRequest, shouldRefreshCache} = interceptor;
+        const fulfillRequestFn = fulfillRequest
+            ? (options: TOptions): Promise<TData> => {
+                  const interceptedResult = fulfillRequest(options);
+                  return interceptedResult != null
+                      ? interceptedResult
+                      : handler.fulfillRequest(options);
```

Comment:
Not sure I understand the question. We allow it to return null so that if we didn't want to intercept that specific request, we can do the default handling instead. It seems plausible to me that we may only want to intercept specific requests and I wanted to provide for that.

Reply:
I see... there might be some requests we want to intercept and others we don't and this allows us to do that.

Reply:
Yeah, exactly.